### PR TITLE
HCX-2 - HCX Protocol APIs

### PR DESCRIPTION
**Pull request by shiva-rakshith**

### CODE REVIEW

File: hcx-apis/src/main/resources/application.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+kafka:
+  topic:
+    payload: ${kafka_payload_topic:local.hcx.request.payload}
+    ingest: ${kafka_ingest_topic:local.hcx.request.ingest}
+
+#hcx headers
+protocol:
+  mandatory:
+    headers: ${protocol_mandatory_headers:x-hcx-sender_code, x-hcx-recipient_code, x-hcx-request_id, x-hcx-correlation_id, x-hcx-timestamp, x-hcx-status, jws_header, jwe_header}
```
Comment (maheshkumargangula) [inline hcx-apis/src/main/resources/application.yml:12]:
```suggestion
protocol:
  headers:
    mandatory: ${protocol_mandatory_headers:x-hcx-sender_code, x-hcx-recipient_code, x-hcx-request_id, x-hcx-correlation_id, x-hcx-timestamp, x-hcx-status, jws_header, jwe_header}
```

Comment (maheshkumargangula) [reply hcx-apis/src/main/resources/application.yml:12]: Please apply the same change for all the other header configurations.

File: hcx-core/hcx-common/src/main/java/org/swasth/common/exception/ClientException.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package org.swasth.common.exception;
+
+public class ClientException extends Exception{
+
+    public ClientException(String message) {
```
Comment (maheshkumargangula) [inline hcx-core/hcx-common/src/main/java/org/swasth/common/exception/ClientException.java:5]: Define errorCode also for the exception.

File: hcx-core/hcx-common/src/main/java/org/swasth/common/exception/ResponseCode.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package org.swasth.common.exception;
+
+public enum ResponseCode {
+
+    OK(200), ACCEPTED(202), CLIENT_ERROR(400), SERVER_ERROR(500), RESOURCE_NOT_FOUND(404), PARTIAL_SUCCESS(207);
```
Comment (maheshkumargangula) [inline hcx-core/hcx-common/src/main/java/org/swasth/common/exception/ResponseCode.java:5]: Remove unused ResponseCodes from this.

**Review by maheshkumargangula — CHANGES_REQUESTED** [review]

File: hcx-apis/src/main/resources/application.yml (hunk 1 above)
Comment (shiva-rakshith) [reply hcx-apis/src/main/resources/application.yml]: Done.

File: hcx-core/hcx-common/src/main/java/org/swasth/common/exception/ClientException.java (hunk 2 above)
Comment (shiva-rakshith) [reply hcx-core/hcx-common/src/main/java/org/swasth/common/exception/ClientException.java:7]: Done

File: hcx-core/hcx-common/src/main/java/org/swasth/common/exception/ResponseCode.java (hunk 3 above)
Comment (shiva-rakshith) [reply hcx-core/hcx-common/src/main/java/org/swasth/common/exception/ResponseCode.java]: Done

File: hcx-apis/src/main/java/org/swasth/hcx/helpers/KafkaEventGenerator.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        constructEvent.put("domain_header",  filterDomainHeaders);
+        constructEvent.put("sender", new HashMap<String,Object>() {{
+            put("participant_code", protectedHeaders.get("x-hcx-sender_code"));
+        }});
+        constructEvent.put("receiver", new HashMap<String,Object>() {{
+            put("participant_code", protectedHeaders.get("x-hcx-recipient_code"));
+        }});
+        constructEvent.put("status", "New");
+        constructEvent.put("log_details", "current status: info");
+        String metadataEvent = new ObjectMapper().writeValueAsString(constructEvent);
```
Comment (maheshkumargangula) [inline hcx-apis/src/main/java/org/swasth/hcx/helpers/KafkaEventGenerator.java:55]: Move this to StringUtils.

**Review by maheshkumargangula — APPROVED** [review]

**Pull request merged**